### PR TITLE
Give the example for custom magics in the docs.

### PR DESCRIPTION
**Pull request by Carreau**
The docs are registering magics by getting the current instance of
IPython and registering directly against it. A better approach is to
actually show the example on how to use the `load_ipython_extension`
function to correctly define a magic.

### CODE REVIEW

File: docs/source/config/custommagics.rst [hunk 1]
```diff
@@ -84,12 +84,17 @@ IPython object:
...
-    ip = get_ipython()
-    # You can register the class itself without instantiating it.  IPython will
-    # call the default constructor on it.
-    ip.register_magics(MyMagics)
+    # running IPython.
+
+    def load_ipython_extension(ipython):
+        """
+        Any nodule file that define a function named load_ipython_extension can
+        be loaded via `load_ext` or be configured to be autoloaded by IPython at
```
Comment (takluyver) [inline docs/source/config/custommagics.rst]: Let's put `%load_ext` to make it clearer that it's a magic.

File: docs/source/config/custommagics.rst [hunk 2]
```diff
@@ -108,11 +113,16 @@ instantiate the class yourself before registration:
...
-    ip.register_magics(magics)
+    def load_ipython_extension(ipython):
+        """
+        Any nodule file that define a function named load_ipython_extension can
+        be loaded via `load_ext` or be configured to be autoloaded by IPython at
+        startup time.
+        """
+        # This class must then be registered with a manually created instance,
+        # since its constructor has different arguments from the default:
+        magics = StatefulMagics(ip, some_data)
```
Comment (takluyver) [inline docs/source/config/custommagics.rst]: ip -> ipython as that's what it's called in the function definition.

**Comment by Carreau** [comment]
Feedback from sprints, it's unclear that load_ipython_extension should be in `__init__.py` and maybe link to flit on how to publish a package.

**Comment by Carreau** [comment]
Thanks !

**Pull request merged**